### PR TITLE
Wireguard: Add wireguard role

### LINKED ISSUE #625: Add a wireguard role
We use Wireguard in the testbed and have a custom playbook for it there. It makes sense to insert this as a Wireguard role after removing the Tailscale role.

https://github.com/osism/testbed/blob/main/environments/custom/playbook-wireguard.yml

### PR DESCRIPTION
**Pull request by ra-rau**
- Add wireguard role
- Add test-role-wireguard to molecule
- This closes #625

Signed-off-by: Ramona Rautenberg <rautenberg@osism.tech>

### CODE REVIEW

**Review by matfechner — APPROVED** [review]

**Review by osfrickler — CHANGES_REQUESTED** [review]

File: .github/workflows/test-role-wireguard.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+name: Test role wireguard
+
+"on":
+  workflow_dispatch:
+  push:
+    paths:
+      - '.github/workflows/test-role-wireguard.yml'
+      - 'molecule/delegated/*'
```
[the remark this answers is not in the document]
Comment (ra-rau) [reply .github/workflows/test-role-wireguard.yml]: Done

File: roles/wireguard/templates/client.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Interface]
+MTU = 1360
+PrivateKey = {{ private_key_client['content']|b64decode }}
+Address = 192.168.48.4/24
```
[the remark this answers is not in the document]
Comment (ra-rau) [reply roles/wireguard/templates/client.conf.j2]: Please check my changes and tell me if it fits.

File: roles/wireguard/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
+---
+- name: Include tasks for ubuntu-bionic
+  ansible.builtin.include_tasks: ubuntu-bionic.yml
+  when: "ansible_distribution_release == 'bionic'"
```
[the remark this answers is not in the document]
Comment (ra-rau) [reply roles/wireguard/tasks/main.yml]: Done

File: roles/wireguard/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+# NOTE: Errors are ignored because the kernel module may not be available
+#       after initial installation until after a reboot.
+- name: Restart wg0 service
+  become: true
+  ansible.builtin.systemd:
+    name: wg-quick@wg0.service
```
Comment (berendt) [inline roles/wireguard/handlers/main.yml:7]: Would prefer to move this into a wireguard_service_name parameter.

File: roles/wireguard/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+- name: Get private key - server
+  become: true
+  ansible.builtin.slurp:
+    src: /etc/wireguard/server.key
+  register: private_key_server
+
+- name: Copy wireguard-client.conf configuration file
+  ansible.builtin.template:
+    src: client.conf.j2
+    dest: "/home/{{ operator_user }}/wireguard-client.conf"
```
Comment (berendt) [inline roles/wireguard/tasks/main.yml:67]: This way we do not have to hardcode the /home/{{ operator_user}}. path.

```
# NOTE: http://stackoverflow.com/questions/33343215/how-to-get-remote-users-home-directory-in-ansible

- name: Get home directory of operator user
  become: true
  ansible.builtin.shell: |
    set -o pipefail
    getent passwd {{ operator_user }} | cut -d: -f6
  args:
    executable: /bin/bash
  changed_when: false
  register: operator_home
```

...

```
dest: "{{ result_create_operator_user.home }}/wireguard-client.conf"
```

**Review by berendt — CHANGES_REQUESTED** [review]

File: roles/wireguard/handlers/main.yml (hunk 4 above)
Comment (ra-rau) [reply roles/wireguard/handlers/main.yml:7]: Done

File: roles/wireguard/tasks/main.yml (hunk 5 above)
Comment (ra-rau) [reply roles/wireguard/tasks/main.yml:67]: wouldn’t it be better to configure the operator home directory in a facts variable and in the operator repository?

File: roles/wireguard/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    group: root
+    mode: 0600
+  notify: Restart wg0 service
+
+# NOTE: Errors are ignored because the kernel module may not be available
+#       after initial installation until after a reboot.
+- name: Start/enable wg-quick@wg0.service service
+  become: true
+  ansible.builtin.systemd:
+    name: wg-quick@wg0.service
```
[the remark this answers is not in the document]
Comment (ra-rau) [reply roles/wireguard/tasks/main.yml:87]: Done

[the remark this answers is not in the document]
Comment (ra-rau) [reply roles/wireguard/tasks/main.yml:87]: Same comment like the one before.

File: roles/wireguard/handlers/main.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+# NOTE: Errors are ignored because the kernel module may not be available
+#       after initial installation until after a reboot.
+- name: Restart wg0 service
+  become: true
+  ansible.builtin.systemd:
+    name: wireguard_service_name
```
Comment (berendt) [inline roles/wireguard/handlers/main.yml:7]: "{{ wireguard_service_name }}"

File: roles/wireguard/tasks/main.yml (hunk 5 above)
Comment (berendt) [reply roles/wireguard/tasks/main.yml:67]: We really only need this on the manager. Therefore, I would not get this generically as a fact.

File: roles/wireguard/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    group: root
+    mode: 0600
+  notify: Restart wg0 service
+
+# NOTE: Errors are ignored because the kernel module may not be available
+#       after initial installation until after a reboot.
+- name: Start/enable wg-quick@wg0.service service
+  become: true
+  ansible.builtin.systemd:
+    name: wireguard_service_name
```
Comment (berendt) [inline roles/wireguard/tasks/main.yml:87]: "{{ wireguard_service_name }}"

**Review by berendt — CHANGES_REQUESTED** [review]

File: roles/wireguard/tasks/main.yml (hunk 5 above)
Comment (ra-rau) [reply roles/wireguard/tasks/main.yml:67]: Done

File: roles/wireguard/handlers/main.yml (hunk 7 above)
Comment (ra-rau) [reply roles/wireguard/handlers/main.yml:7]: Done

**Pull request merged**